addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -87,11 +88,11 @@ def _check_move_configuration(self, move):
             error_message.append(_("Invoice number should not be more than 16 characters"))
         all_base_tags = self._get_l10n_in_base_tags()
         for line in move.invoice_line_ids.filtered(lambda line: line.display_type not in ('line_note', 'line_section', 'rounding')):
-            if line.price_subtotal < 0:
-                # Line having a negative amount is not allowed.
-                if not move._l10n_in_edi_is_managing_invoice_negative_lines_allowed():
-                    raise ValidationError(_("Invoice lines having a negative amount are not allowed to generate the IRN. "
-                                  "Please create a credit note instead."))
+            if line.price_subtotal < 0 and not discount_line_posted:
+                move.message_post(
+                        body="The negative lines will be repartitioned as a discount on the positive lines."
```

Comment:
Improve message into "Negative lines will be decreased from positive invoice lines having the same taxes and HSN code".